eta/core/serial.py

```
@@ -1124,17 +1113,477 @@ def _filter_elements(self, filters, match):
+
+    This class cannot be instantiated directly. Instead a subclass should
+    be created for each type of element to be stored. Subclasses MUST set the
+    following members:
+        -  `_ELE_CLS`: the class of the element stored in the container
+
+    In addition, sublasses MAY override the following members:
+        - `_ELE_CLS_FIELD`: the name of the private attribute that will store
+            the class of the elements in the container
+        - `_ELE_ATTR`: the name of the attribute that will store the elements
+            in the container
+
+    Examples:
+        ```
+        import eta.core.geometry as etag
+
+        point = etag.LabeledPoint("origin", etag.RelativePoint(0, 0))
+        points = etag.BigLabeledPointContainer("/tmp/BigLabeledPointContainer")
+
+        # Immediately writes the LabeledPoint to disk
+        points.add(point)
+
+        # Reads the LabeledPoint from disk
+        print(points[0])
+
+        # Only the index of the BigContainer is serialized
+        print(points)
+        ```
+    '''
+
+    def __init__(self, backing_dir, **kwargs):
+        '''Creates a BigContainer instance.
+
+        Args:
+            backing_dir: the backing directory in which the elements are/will
+                be stored
+            <elements>: an optional list of uuids for elements in the
+                container. The appropriate name of this keyword argument is
+                determined by the `_ELE_ATTR` member of the BigContainer
+                subclass
+
+        Raises:
+            ContainerError: if there was an error while creating the container
+        '''
+        self._validate_cls()
+
+        self._backing_dir = None
+        self._set_backing_dir(backing_dir)
+
+        if self._ELE_ATTR in kwargs:
+            etau.ensure_dir(self.backing_dir)
+            elements = kwargs[self._ELE_ATTR]
+        else:
+            etau.ensure_empty_dir(self.backing_dir)
```

Note that I removed the code that would try to automatically populate a BigContainer based on the contents of the backing directory. I did this because I think `BigContainer(backing_dir)` should throw an error if `backing_dir` is not empty.

The user can use `BigContainer.from_paths` or `BigContainer.from_dir` if they want to automatically populate a `BigContainer` from disk